ios/tunnel/tunnel_api.go

```diff
@@ -64,8 +64,18 @@ func RunAgent(args ...string) error {
 		return fmt.Errorf("RunAgent: failed to get executable path: %w", err)
 	}
 
-	cmd := exec.Command(ex, append([]string{"tunnel", "start"}, args...)...)
+	var cmd *exec.Cmd
+
+	if mode == "kernel" {
+		cmd = exec.Command(ex, append([]string{"tunnel", "start"}, args...)...)
+	}
+
+	if mode == "user" {
+		cmd = exec.Command(ex, append([]string{"tunnel", "start", "--userspace"}, args...)...)
+	}
```

Comment:
lets verify that we get one of the expected parameters, otherwise it would panic
```suggestion
	var cmd *exec.Cmd
	switch mode {
	case "kernel":
		cmd = exec.Command(ex, append([]string{"tunnel", "start"}, args...)...)
	case "user":
		cmd = exec.Command(ex, append([]string{"tunnel", "start", "--userspace"}, args...)...)
	default:
		return fmt.Errorf("RunAgent: unknown mode: %s. Only 'kernel' and 'user' are supported", mode)
	}
```

Reply:
I've modified with the suggestion. Initial code implemented like the old code for checking only 'yes' value and if only for that value it was calling RunAgent(). Similar, initial code was calling RunAgent() only for 'kernel' and 'user' values. However what @dmissmann suggested is better practice to handle panic independent of the caller. Modified accoringly.